javalib/src/main/scala/java/util/concurrent/TimeUnit.scala

```diff
@@ -0,0 +1,99 @@
+package java.util.concurrent
+
+sealed abstract class TimeUnit(private val id: Int)
+    extends Serializable
+    with Comparable[TimeUnit] {
+  import TimeUnit._
+  override def compareTo(o: TimeUnit): Int =
+    id.compareTo(o.id)
+
+  protected def makeNext: Long => Long
+  protected def makePrev: Long => Long
+  protected def next: TimeUnit
+  protected def prev: TimeUnit
+
+  def convert(sourceDuration: Long, sourceUnit: TimeUnit): Long =
+    if (id > sourceUnit.id) convert(makeNext(sourceDuration), sourceUnit.next)
+    else if (id == sourceUnit.id) sourceDuration
+    else convert(makePrev(sourceDuration), sourceUnit.prev)
```

Comment:
Another data point is an implementation from Scala.js: https://github.com/scala-js/scala-js/blob/master/javalib/src/main/scala/java/util/concurrent/TimeUnit.scala